src/Uno.UI/ts/Windows/Security/Authentication/Web/WebAuthenticationBroker.ts

```diff
@@ -0,0 +1,202 @@
+﻿namespace Windows.Security.Authentication.Web {
+	export class WebAuthenticationBroker {
+
+		public static getReturnUrl() {
+			return window.location.origin;
+		}
+
+		public static authenticateUsingIframe(
+			iframeId: string,
+			urlNavigate: string,
+			urlRedirect: string,
+			timeout: number
+		) {
+			return new Promise<string>((ok, err) => {
+				let iframe: HTMLIFrameElement;
+
+				if (iframeId) {
+					iframe = document.getElementById(iframeId) as HTMLIFrameElement;
+				}
+
+				if (!iframe) {
+					iframe = document.createElement("iframe");
+					iframe.style.opacity = "0";
+					iframe.style.pointerEvents = "none";
+					document.body.append(iframe);
+				}
+
+				const terminate = () => {
+					iframe.removeEventListener("load", onload);
+					iframe.src = "about:blank";
+					if (!iframeId) {
+						iframe.remove();
+					}
+				}
+
+				const onload = () => {
+					if (!iframe.contentDocument) {
+						return; // can't access right now
+					}
+
+					const currentUrl = iframe.contentDocument.URL;
+					console.log("iframe src=" + currentUrl);
+
+					if (currentUrl.indexOf(urlRedirect) === 0) {
+						terminate();
+						ok(`success|${currentUrl}`);
+					}
+				};
+
+				iframe.addEventListener("load", onload);
+
+				iframe.src = urlNavigate;
+			})
+				;
+		}
+
+		public static authenticateUsingWindow(
+			urlNavigate: string,
+			urlRedirect: string,
+			title: string,
+			popUpWidth: number,
+			popUpHeight: number,
+			timeout: number): Promise<string> {
+
+			let win: Window = null;
+			let timerSubscription: number = null;
+
+			return new Promise<string>((ok, err) => {
+
+				let finished = false;
+				const close = () => {
+					if (win) {
+						win.close();
+						win = null;
+					}
+
+					if (timerSubscription) {
+						window.clearInterval(timerSubscription);
+						timerSubscription = null;
+					}
+					if (!finished) {
+						err("Incompleted");
```

Comment:
```suggestion
						err("Incomplete");
```

What does "Incomplete" mean ? Does the developer need to do something about it ?